tag/nitag.yml

```diff
@@ -0,0 +1,1814 @@
+        type: string
+      type:
+        type: string
+        enum: *TYPES
+    example:
+      type: DOUBLE
+      value: "3.14"
+  TagPathWithCurrentAndAggregateValue:
+    description: Tag path with its current value and aggregates
+    type: object
+    title: Tag Path Current And Aggregate Value
+    properties:
+      path:
+        type: string
+      current:
+        $ref: '#/definitions/TagCurrentValue'
+      aggregates:
+        $ref: '#/definitions/TagAggregates'
+    example:
+      path: foo
+      current:
+        value:
+          type: DOUBLE
+          value: "3.14"
+      aggregates:
+        sum: "10.5"
+        min: "0.0"
+        max: "5.0"
+        mean: "2.0"
+        count: 5
+  TagCurrentValue:
+    description: Current value of a tag
+    type: object
+    title: Tag Current Value
+    properties:
+      value:
+        $ref: '#/definitions/TagValue'
+      timestamp:
+        description: Timestamp associated with the value
+        type: string
+        format: date-time
+    example:
+      value:
+        type: DOUBLE
+        value: "3.14"
+      timestamp: '2018-09-04T18:45:08Z'
+  TagCurrentAndAggregateValue:
+    type: object
+    title: Tag Current And Aggregate Value
+    properties:
+      current:
+        $ref: '#/definitions/TagCurrentValue'
+      aggregates:
+        $ref: '#/definitions/TagAggregates'
+    example:
+      current:
+        value:
+          type: DOUBLE
+          value: "3.14"
+        timestamp: '2018-09-04T18:45:08Z'
+      aggregates:
+        sum: "10.5"
+        min: "0.0"
+        max: "5.0"
+        mean: "2.0"
+        count: 5
+  TagAggregates:
+    description: Aggregate values of a tag
+    type: object
+    title: Tag Aggregate Value
+    properties:
+      sum:
+        description: Sum aggregate value
+        type: number
+      min:
+        description: Minimum aggregate value
+        type: number
+      max:
+        description: Maximum aggregate value
+        type: number
+      count:
+        description: Count aggregate value
+        type: integer
+      avg:
+        description: Average aggregate value
+        type: number
+    example:
+      sum: "10.5"
+      min: "0.0"
+      max: "5.0"
+      mean: "2.0"
+      count: 5
+  Tag:
+    description: Object containing a tag's metadata, excluding current and aggregate values
+    type: object
+    title: Tag
+    properties:
+      type:
+        description: Type
+        type: string
+        enum: *TYPES
+      properties:
+        type: object
+      path:
+        type: string
+      keywords:
+        type: array
+        items:
+          type: string
+      collectAggregates:
+        type: boolean
+    example:
+      type: DOUBLE
+      properties:
+        key1: value1
+        key2: value2
+      path: foo
+      keywords: [fooKeyword, barKeyword]
+      collectAggregates: true
+  TagUpdate:
+    description: Object containing a tag's path, and one or more updates to be applied to it
```

Comment:
Are updates values, or separate actions?

Reply:
Updates are values - a list of tag values with timestamps which should be applied to a tag.

Reply:
What do you think about "and one or more values to update the tag with at times you specify?" It's longer, but clarifies the level of control the user has here.

Reply:
I like that a lot more, thanks!